File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task13.cpp

#include <iostream>
#include <vector>
#include <string>
#include <cstddef>

// Function to validate if a 10-minute walk returns to starting point
// Security: Validates input size and characters to prevent undefined behavior
// Returns true if walk takes exactly 10 minutes and returns to origin
bool isValidWalk(const std::vector<std::string>& walk) {
    // Validate input: must be exactly 10 directions for 10 minutes
    // Security: Check size to prevent processing invalid input
    if (walk.size() != 10) {
        return false;
    }
    
    // Track position on grid: x (east/west), y (north/south)
    int x = 0;
    int y = 0;
    
    // Process each direction
    for (size_t i = 0; i < walk.size(); ++i) {
        // Security: Validate each direction string is not empty
        if (walk[i].empty()) {
            return false;
        }
        
        // Security: Validate only single character directions
        if (walk[i].length() != 1) {
            return false;
        }
        
        char direction = walk[i][0];
        
        // Security: Validate direction is one of expected values
        // Update position based on validated direction
        switch (direction) {
            case 'n':
                y++;
                break;
            case 's':
                y--;
                break;
            case 'e':
                x++;
                break;
            case 'w':
                x--;
                break;
            default:
                // Security: Reject invalid direction characters
                return false;
        }
    }
    
    // Check if we returned to starting point (0, 0)
    return (x == 0 && y == 0);
}

int main() {
    // Test case 1: Valid 10-minute walk returning to origin
    std::vector<std::string> test1 = {"n", "s", "n", "s", "n", "s", "n", "s", "n", "s"};
    std::cout << "Test 1: " << (isValidWalk(test1) ? "true" : "false") << std::endl;
    
    // Test case 2: Valid walk with all directions
    std::vector<std::string> test2 = {"n", "e", "s", "w", "n", "e", "s", "w", "n", "s"};
    std::cout << "Test 2: " << (isValidWalk(test2) ? "true" : "false") << std::endl;
    
    // Test case 3: Too short (less than 10 minutes)
    std::vector<std::string> test3 = {"n", "s", "e", "w"};
    std::cout << "Test 3: " << (isValidWalk(test3) ? "true" : "false") << std::endl;
    
    // Test case 4: Correct length but doesn't return to origin
    std::vector<std::string> test4 = {"n", "n", "n", "n", "n", "n", "n", "n", "n", "n"};
    std::cout << "Test 4: " << (isValidWalk(test4) ? "true" : "false") << std::endl;
    
    // Test case 5: Too long (more than 10 minutes)
    std::vector<std::string> test5 = {"n", "s", "n", "s", "n", "s", "n", "s", "n", "s", "n", "s"};
    std::cout << "Test 5: " << (isValidWalk(test5) ? "true" : "false") << std::endl;
    
    return 0;
}

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task13.py
def is_valid_walk(walk):
    if walk is None or len(walk) != 10:
        return False
    x = y = 0
    for c in walk:
        if c == 'n':
            y += 1
        elif c == 's':
            y -= 1
        elif c == 'e':
            x += 1
        elif c == 'w':
            x -= 1
        else:
            return False
    return x == 0 and y == 0

if __name__ == "__main__":
    w1 = ['n','s','n','s','e','w','e','w','n','s']  # true
    w2 = ['n','s','e']  # false (length != 10)
    w3 = ['n','n','n','n','n','s','s','s','e','w']  # false
    w4 = ['e','w','e','w','n','s','n','s','n','s']  # true
    w5 = ['n','n','n','n','n','n','n','n','n','n']  # false

    print(is_valid_walk(w1))
    print(is_valid_walk(w2))
    print(is_valid_walk(w3))
    print(is_valid_walk(w4))
    print(is_valid_walk(w5))